files/bootstrap.sh

```diff
@@ -586,26 +554,23 @@ if [[ "$CONTAINER_RUNTIME" = "containerd" ]]; then
 
     sudo mkdir -p /etc/containerd
     sudo mkdir -p /etc/cni/net.d
-    mkdir -p /etc/systemd/system/containerd.service.d
-    cat <<EOF > /etc/systemd/system/containerd.service.d/10-compat-symlink.conf
-[Service]
-ExecStartPre=/bin/ln -sf /run/containerd/containerd.sock /run/dockershim.sock
-EOF
     if [[ -n "$CONTAINERD_CONFIG_FILE" ]]; then
         sudo cp -v $CONTAINERD_CONFIG_FILE /etc/eks/containerd/containerd-config.toml
     fi
     echo "$(jq '.cgroupDriver="systemd"' $KUBELET_CONFIG)" > $KUBELET_CONFIG
     sudo sed -i s,SANDBOX_IMAGE,$PAUSE_CONTAINER,g /etc/eks/containerd/containerd-config.toml
-    sudo cp -v /etc/eks/containerd/containerd-config.toml /etc/containerd/config.toml
-    sudo cp -v /etc/eks/containerd/sandbox-image.service /etc/systemd/system/sandbox-image.service
+    # Check if the same PAUSE container was used when building the AMI
+    # If different, then restart containerd w/ proper sandbox config
+    if ! cmp -s /etc/eks/containerd/containerd-config.toml /etc/containerd/config.toml; then
```

Comment:
can we make this check a little more specific to the pause image? could a user make unrelated changes to these file(s) and trigger this unintentionally?

Reply:
Updated comment